src/log.h

```diff
@@ -139,6 +139,64 @@ class FileLogOutput : public ICombinedLogOutput {
 	std::ofstream m_stream;
 };
 
+struct LogEntry {
+	std::string timestamp;
+	std::string thread_name;
+	std::string text;
+	// "timestamp: level[thread_name]: text"
+	std::string combined;
+
+	LogEntry(std::string timestamp, std::string thread_name, std::string text, std::string combined) :
+		timestamp(std::move(timestamp)),
+		thread_name(std::move(thread_name)),
+		text(std::move(text)),
+		combined(std::move(combined)) { }
+};
+
+class CaptureLogOutput : public ILogOutput {
```

Comment:
Should be moved to the test section. This header should stay small because it's included in almost every other file.

`LogEntry` too.

Reply:
If this is a concern, I believe I could actually remove Logger, and all the ILogOutput classes from `log.h` into a separate `logconfig.h`. I will do that in another PR first.

Reply:
For reference #15258